[snappy] build failure

**Host Environment**
- OS: Centos7
- Compiler: revision

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install srpc`

**Failure logs**     
```
usr/downloads/vcpkg/buildtrees/snappy/src/1.1.9-3644d789ea.clean/snappy.cc:1017:8: error: inlining failed in call to always_inline ‘size_t snappy::AdvanceToNextTag(const uint8_t**, size_t*)’: function body can be overwritten at link time
/usr/downloads/vcpkg/buildtrees/snappy/src/1.1.9-3644d789ea.clean/snappy.cc:1097:53: error: called from here
         size_t tag_type = AdvanceToNextTag(&ip, &tag);
                                                     ^
/usr/downloads/vcpkg/buildtrees/snappy/src/1.1.9-3644d789ea.clean/snappy.cc: In function ‘std::pair<const unsigned char*, long int> snappy::DecompressBranchless(const uint8_t*, const uint8_t*, ptrdiff_t, T, ptrdiff_t) [with T = char*; uint8_t = unsigned char; ptrdiff_t = long int]’:
/usr/downloads/vcpkg/buildtrees/snappy/src/1.1.9-3644d789ea.clean/snappy.cc:1017:8: error: inlining failed in call to always_inline ‘size_t snappy::AdvanceToNextTag(const uint8_t**, size_t*)’: function body can be overwritten at link time
 size_t AdvanceToNextTag(const uint8_t** ip_p, size_t* tag) {
        ^
/usr/downloads/vcpkg/buildtrees/snappy/src/1.1.9-3644d789ea.clean/snappy.cc:1097:53: error: called from here
         size_t tag_type = AdvanceToNextTag(&ip, &tag);
                                                     ^
ninja: build stopped: subcommand failed.
```
logs above in /vcpkg/buildtress/snappy/install-x64-linux-rel-out.log when I tried to  install srpc.

**Additional context** 
```
-- Building x64-linux-dbg
-- Building x64-linux-rel
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:157 (message):
    Command failed: /usr/local/cmake/bin/cmake --build . --config Release --target install -- -v -j5
    Working Directory: /usr/downloads/vcpkg/buildtrees/snappy/x64-linux-rel
    See logs for more information:
      /usr/downloads/vcpkg/buildtrees/snappy/install-x64-linux-rel-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:102 (vcpkg_execute_build_process)
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:41 (vcpkg_cmake_build)
  ports/snappy/portfile.cmake:17 (vcpkg_cmake_install)
  scripts/ports.cmake:142 (include)


Error: Building package snappy:x64-linux failed with: BUILD_FAILED
```
logs above are logs on terminal

--- a/ports/snappy/portfile.cmake
+++ b/ports/snappy/portfile.cmake
@@ -2,7 +2,7 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO google/snappy
     REF ${VERSION}
-    SHA512 3578597f1d4ec09104ce0296b825b013590351230dfe56c635081fd282ce7a13a34caf2c283ac77bd24065e2d27af6db068d1f84b98cec2fd39a0e37a0d77070
+    SHA512 e7290d79ddd45605aafd02cba9eaa32309c94af04f137552a97a915c391f185dccab9b7b21a01b28f3f446be420232c3c22d91c06e0be6e1e2e32d645174798c
     HEAD_REF master
     PATCHES
         fix_clang-cl_build.patch

--- a/ports/snappy/vcpkg.json
+++ b/ports/snappy/vcpkg.json
@@ -1,7 +1,6 @@
 {
   "name": "snappy",
-  "version": "1.1.10",
-  "port-version": 1,
+  "version": "1.2.1",
   "description": "A fast compressor/decompressor.",
   "homepage": "https://github.com/google/snappy",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -8229,8 +8229,8 @@
       "port-version": 2
     },
     "snappy": {
-      "baseline": "1.1.10",
-      "port-version": 1
+      "baseline": "1.2.1",
+      "port-version": 0
     },
     "sndfile": {
       "baseline": "0",

--- a/versions/s-/snappy.json
+++ b/versions/s-/snappy.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "9672ae749ed648326f88d504ae03872883a526e4",
+      "version": "1.2.1",
+      "port-version": 0
+    },
+    {
       "git-tree": "52d0b0f59c6f6e765c7f8df9e075ff7b90552c1e",
       "version": "1.1.10",
       "port-version": 1

